### ISSUE TITLE
Consistent periodic 403's

### ISSUE THREAD

**Issue by shaun-blake** [vfarcic/docker-flow-proxy#185]
I have a few copies of two different stacks running along with DFP on a single node swarm.

For one of the stacks, about 1 out of every 3 requests to any of the services (1 front-end and 2 back-end services) will result in a 503 from DFP. This gif illustrates refreshing one of the services while following the logs.

![403-problem](https://cloud.githubusercontent.com/assets/232912/24059300/b3b9fee4-0af2-11e7-9345-acc09fae65ed.gif)

For the other application I don't see the same problem.

To add a little more to my confusion - the two stacks each have a service based on the same image. In one stack I don't see the issue for that service, while I do in the other.

### Stacks

DFP - I have `replicas: 2` but I've scaled it to 1 to make watching the logs easier
```yml
version: "3.1"

services:

  proxy:
    image: vfarcic/docker-flow-proxy:1.274
    ports:
      - 80:80
      - 443:443
    networks:
      - proxy
    environment:
      - LISTENER_ADDRESS=swarm-listener
      - SKIP_ADDRESS_VALIDATION=true
      - MODE=swarm
      - DEBUG=true
    secrets:
      - dfp_stats_user
      - dfp_stats_pass
    deploy:
      replicas: 2
    logging:
      driver: json-file
      options:
        max-size: "5m"
        max-file: "2"

  swarm-listener:
    image: vfarcic/docker-flow-swarm-listener:1.22
    networks:
      - proxy
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
    environment:
      - DF_NOTIFY_CREATE_SERVICE_URL=http://proxy:8080/v1/docker-flow-proxy/reconfigure
      - DF_NOTIFY_REMOVE_SERVICE_URL=http://proxy:8080/v1/docker-flow-proxy/remove
    deploy:
      placement:
        constraints: [node.role == manager]
    logging:
      driver: json-file
      options:
        max-size: "5m"
        max-file: "2"

networks:
  proxy:
    external: true
secrets:
  dfp_stats_user:
    external: true
  dfp_stats_pass:
    external: true
```

My application stack (slightly redacted):
```yml
version: '3.1'
services:
  service-in-common:
    image: registry.gitlab.com/something/service-in-common:version
    deploy:
      labels:
        com.df.notify: "true"
        com.df.distribute: "true"
        com.df.servicePath: "/ohrb/${BRANCH_NAME:-test}/service-in-common"
        com.df.reqPathSearch: "/ohrb/${BRANCH_NAME:-test}/service-in-common"
        com.df.reqPathReplace: "/service-in-common"
        com.df.port: "8080"
    healthcheck:
      test: ["CMD", "wget", "-O-", "http://localhost:8080/service-in-common/status"]
      interval: 5s
      timeout: 10s
      retries: 10
    networks:
      - proxy
      - default
  server:
    image: registry.gitlab.com/something/server:${IMAGE_TAG:-TEST_ENVIRONMENT}
    build:
      context: ../../server
    deploy:
      labels:
        com.df.notify: "true"
        com.df.distribute: "true"
        com.df.servicePath: "/ohrb/${BRANCH_NAME:-test}/server"
        com.df.reqPathSearch: "/ohrb/${BRANCH_NAME:-test}/server"
        com.df.reqPathReplace: "/"
        com.df.port: "8080"
    healthcheck:
      test: ["CMD", "wget", "-O-", "http://localhost:8080"]
      interval: 5s
      timeout: 10s
      retries: 30
    networks:
      - proxy
      - default
  client:
    image: registry.gitlab.com/something/client:${IMAGE_TAG:-TEST_ENVIRONMENT}
    build:
      context: ../../client
    deploy:
      labels:
        com.df.notify: "true"
        com.df.distribute: "true"
        com.df.servicePath: "/ohrb/${BRANCH_NAME:-test}/client"
        com.df.reqPathSearch: "/ohrb/${BRANCH_NAME:-test}/client"
        com.df.reqPathReplace: "/"
        com.df.port: "80"
    healthcheck:
      test: ["CMD", "wget", "-O-", "http://localhost"]
      interval: 5s
      timeout: 10s
      retries: 30
    networks:
      - proxy
      - default
    command: nginx -g "daemon off;"

networks:
  proxy:
    external: true
  default:

```

**Comment by shaun-blake** [comment]
Actually, just a couple of hours ago our docker host was patched and restarted.  Since then I haven't had the issue, so I don't have logs from when it happened.  You may be able to see them from the animated git in the description, though.

Feel free to close the issue.  I could reopen it if happens again.  I thought I'd better open the issue since even though it's not happening anymore in case it is an actual problem (since I already had all the information down).

**Comment by vfarcic** [comment]
Thanks for recording this issue. I'll close it for now and keep an eye on feedback from other users. If anyone else experiences a similar problem or if it happens to you again, we'll reopen it and work on a resolution.

**Comment by IlyasDeckers** [comment]
I am facing exactly the same behavior. I've dealt with it before, the problem back then was that I was running 2 swarm listeners by mistake, but now it happened again when building a new cluster from the ground up.

I noticed that when I scaled up the proxy service to 2 the issue was gone. I would like to provide some logs, but I can't seem to find anything that is related to this issue. 

Later on I removed the proxy service and redeployed it to the swarm, from this point I did not have any issues
